qiskit/visualization/latex.py

```diff
@@ -345,690 +317,319 @@ def _get_beamer_page(self):
-
-                            elif nm == "cswap":
-                                if cond_1 == '0':
-                                    self._latex[pos_1][column] = "\\ctrlo{" + str(
-                                        pos_2 - pos_1) + "}"
-                                elif cond_1 == '1':
-                                    self._latex[pos_1][column] = "\\ctrl{" + str(
-                                        pos_2 - pos_1) + "}"
-                                self._latex[pos_2][column] = "\\qswap"
-                                self._latex[pos_3][column] = \
-                                    "\\qswap \\qwx[" + str(pos_2 - pos_3) + "]"
-                            else:
-                                start_pos = min([pos_1, pos_2, pos_3])
-                                stop_pos = max([pos_1, pos_2, pos_3])
-                                if stop_pos - start_pos >= 3:
-                                    delta = stop_pos - start_pos
-                                    self._latex[start_pos][column] = ("\\multigate{%s}{%s}" %
-                                                                      (delta, nm))
-                                    for i_pos in range(start_pos + 1, stop_pos + 1):
-                                        self._latex[i_pos][column] = ("\\ghost{%s}" %
-                                                                      nm)
-                                else:
-                                    self._latex[pos_1][column] = ("\\multigate{2}{%s}" %
-                                                                  nm)
-                                    self._latex[pos_2][column] = ("\\ghost{%s}" %
-                                                                  nm)
-                                    self._latex[pos_3][column] = ("\\ghost{%s}" %
-                                                                  nm)
-
-                    elif len(qarglist) > 3:
-                        nbits = len(qarglist)
-                        pos_array = [self.img_regs[qarglist[0]]]
-                        for i in range(1, nbits):
-                            pos_array.append(self.img_regs[qarglist[i]])
-                        pos_start = min(pos_array)
-                        pos_stop = max(pos_array)
-                        self._latex[pos_start][column] = ("\\multigate{%s}{%s}" %
-                                                          (nbits - 1, nm))
-                        for pos in range(pos_start + 1, pos_stop + 1):
-                            self._latex[pos][column] = ("\\ghost{%s}" % nm)
-
-                elif op.name == "measure":
-                    if (len(op.cargs) != 1
-                            or len(op.qargs) != 1
-                            or op.op.params):
-                        raise exceptions.VisualizationError("bad operation record")
+                elif op.name in ['barrier', 'snapshot', 'load', 'save', 'noise']:
```

Comment:
I think it would be great to slowing get out of depending of `op.name` and move towards `isinstanceof()`. This applies to the other drawer probably. Maybe not in this PR but in a follow up later.

Reply:
Went ahead and did this almost everywhere in the latex drawer in https://github.com/Qiskit/qiskit-terra/pull/5855/commits/9d58cd45651dd1d59c2a4462052bb5772026b802.